cmd/apptable/src/lib.rs

```diff
@@ -24,7 +25,10 @@ struct ApptableArgs {
 }
 
 #[rustfmt::skip::macros(println, bail)]
-fn apptablecmd(hubris: &mut HubrisArchive, subargs: &[String]) -> Result<()> {
+fn apptablecmd(context: &mut humility::ExecutionContext) -> Result<()> {
+    let Subcommand::Other(subargs) = context.cli.cmd.as_ref().unwrap();
```

Comment:
I wanted to do this, but the problem is that it doesn't work.

Consider this patch here:

```diff
    //let Subcommand::Other(subargs) = context.cli.cmd.as_ref().unwrap();
    //let hubris = context.archive.as_mut().unwrap();
    let subargs = context.subargs();
    let hubris = context.hubris();
```

Here's those methods:

```rust
    pub fn subargs(&self) -> &Vec<String> {
        let Subcommand::Other(subargs) = self.cli.cmd.as_ref().unwrap();
        subargs
    }

    pub fn hubris(&mut self) -> &mut HubrisArchive {
        self.archive.as_mut().unwrap()
    }
```

Pretty straightforward. The problem is that this doesn't compile:

```
error[E0502]: cannot borrow `*context` as mutable because it is also borrowed as immutable
  --> cmd\apptable\src\lib.rs:32:18
   |
31 |     let subargs = context.subargs();
   |                   ----------------- immutable borrow occurs here
32 |     let hubris = context.hubris();
   |                  ^^^^^^^^^^^^^^^^ mutable borrow occurs here
33 |
34 |     let subargs = ApptableArgs::try_parse_from(subargs)?;
   |                                                ------- immutable borrow later used here
```

Once you encapsulate them, borrowcheck no longer can say that they're disjoint, because you're not operating on the parts, but instead on the whole thing.

Reply:
...yeah, I'd leave them as fields for this reason.